### ISSUE TITLE
[ConfigurationService] Service Components are not cleaned up at shutdown

### ISSUE THREAD

**Issue by svilenvul** [eclipse/kura#2190]
Hello, I would like to report a bug that is related with the ConfigurationService and more specifically with the lifecycle of the Service Components that it creates.

  - Distribution
    - OSGi Runtime (none Equinox, OSGi data is NOT cleaned at restart)
    - Kura version affected (Kura 3.2.0): 

The issue can be reproduced:

- [osgi.clean](https://help.eclipse.org/neon/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fmisc%2Fruntime-options.html) should be set to false (or equivalent in none Equinox OSGi runtime);
- Cloud Service, Data Service configuration is saved in a snapshot file;
- OSGi runtime is restarted several times;
- check the number of registered service components, it will grow linear with each restart :);

It is a bit hard to follow the flow, but I guess that ConfigurationService.createFactoryConfiguration() is called when an old snapshot is being loaded, but the created service components are not removed 
 with ConfigurationService.deleteFactoryConfiguration.

This issue becomes critical when H2DbService is instantiated several times and multiple instances try to open a connection to the same URL.

Partly related to https://github.com/eclipse/kura/pull/2027/files.

**Comment by MMaiero** [comment]
Hi, 
at the moment the osgi.clean=false modality is not supported by Kura.

It's in the roadmap an exploration of this working configuration but it will imply heavy changes in the configuration service, as you correctly pointed out.

Best regards,
Matteo
